### PR TITLE
DRAFT: Swappable rendition backends

### PR DESCRIPTION
**Pull request by ababic**
TBC

### CODE REVIEW

File: wagtail/images/rendition_backends/model.py [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    A mixin for rendition backends that use a Django model to store and
+    recall rendtions for images.
+    """
+
+    @property
+    def renditions_cache(self) -> BaseCache | None:
+        try:
+            return caches["renditions"]
+        except InvalidCacheBackendError:
+            return None
```
Comment (zerolab) [inline wagtail/images/rendition_backends/model.py:32]: note: as of #10612 we always cache renditions

**Comment by zerolab** [comment]
@ababic discussed this in the core meeting today. Would be nice to have an example non-Wagtail backends, if possible

**Comment by ababic** [comment]
Cheers @zerolab! I have some WIP code I can add.

**Comment by ababic** [comment]
@zerolab We've ruled out "external rendition generation" on the project I'm currently working on, so am less likely to find project time I can 'invest back' here. Though, I am personally interested in playing with Cantaloupe, so will aim to but a `CantaloupeRenditionBackend` together in due course.

With WebP and AVIF support, it turns out Willow really quite a strong option :)

**Comment by lb-** [comment]
We will create a new discussion out of the content of this PR, that way we can close the PR as @ababic is unable to work on this.

From here we can get some feedback from the community as to whether there is buy in for this.

@allcaps - are you able to create one please.

**Comment by allcaps** [comment]
I've created a discussion https://github.com/wagtail/wagtail/discussions/12513

@ababic Thank you for your fresh ideas. Is the discussion text a good reflection of the intended feature? I'd love your input and hear more about your motivation / use-case!

**Pull request closed without merging**